File: doc-examples/r2dbc-example-java/src/main/java/example/Book.java
package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
@MappedEntity
public class Book {
    @Id
    @GeneratedValue
    private Long id;
    private final String title;
    private final int pages;
    @Relation(Relation.Kind.MANY_TO_ONE)
    private final Author author;

    public Book(String title, int pages, Author author) {
        this.title = title;
        this.pages = pages;
        this.author = author;
    }

    public Author getAuthor() {
        return author;
    }

    public String getTitle() {
        return title;
    }

    public int getPages() {
        return pages;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
}

File: doc-examples/r2dbc-example-kotlin/src/main/kotlin/example/Book.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation
import io.micronaut.serde.annotation.Serdeable

@Serdeable
@MappedEntity
data class Book(
        val title: String,
        val pages: Int,
        @Relation(Relation.Kind.MANY_TO_ONE)
        val author: Author) {
    @Id
    @GeneratedValue
    var id: Long? = null
}
